File: svws-core/src/main/java/de/nrw/schule/svws/core/utils/stundenplanblockung/StundenplanblockungManagerKopplung.java
package de.nrw.schule.svws.core.utils.stundenplanblockung;

import java.util.Vector;
import jakarta.validation.constraints.NotNull;

/**
 * 
 * @author Benjamin A. Bartsch
 */
public class StundenplanblockungManagerKopplung {

	/** Die Datenbank-ID der Kopplung. */
	private long _id;

	/** Das Kürzel der Kopplung. Beispielsweise '7RE'. */
	private @NotNull String _kuerzel = "";

	/** Alle Lerngruppen der Kopplungen. */
	private @NotNull Vector<StundenplanblockungManagerLerngruppe> _lerngruppen = new Vector<>();

	/**
	 * Erzeugt eine neue Kopplung.
	 * 
	 * @param pKopplungID  Die Datenbank-ID der Kopplung.
	 * @param pKuerzel     Das Kürzel der Kopplung.
	 */
	public StundenplanblockungManagerKopplung(long pKopplungID, @NotNull String pKuerzel) {
		_id = pKopplungID;
		_kuerzel = pKuerzel;
	}
	
	/**
	 * Liefert die Datenbank-ID der Kopplung.
	 * 
	 * @return Die Datenbank-ID der Kopplung.
	 */
	public long getID() {
		return _id;
	}
	
	/**
	 * Setzt das Kürzel der Kopplung.
	 * 
	 * @param pKuerzel  Das neue Kürzel der Kopplung.
	 */
	public void setKuerzel(@NotNull String pKuerzel) {
		_kuerzel = pKuerzel;
	}
}

File: svws-webclient/src/ts-lib/ts/src/core/utils/stundenplanblockung/StundenplanblockungManagerKopplung.ts
import { JavaObject, cast_java_lang_Object } from '../../../java/lang/JavaObject';
import { JavaString, cast_java_lang_String } from '../../../java/lang/JavaString';
import { StundenplanblockungManagerLerngruppe, cast_de_nrw_schule_svws_core_utils_stundenplanblockung_StundenplanblockungManagerLerngruppe } from '../../../core/utils/stundenplanblockung/StundenplanblockungManagerLerngruppe';
import { Vector, cast_java_util_Vector } from '../../../java/util/Vector';

export class StundenplanblockungManagerKopplung extends JavaObject {

	private _id : number = 0;

	private _kuerzel : String = "";

	private _lerngruppen : Vector<StundenplanblockungManagerLerngruppe | null> = new Vector();


	/**
	 * Erzeugt eine neue Kopplung.
	 * 
	 * @param pKopplungID  Die Datenbank-ID der Kopplung.
	 * @param pKuerzel     Das Kürzel der Kopplung.
	 */
	public constructor(pKopplungID : number, pKuerzel : String) {
		super();
		this._id = pKopplungID;
		this._kuerzel = pKuerzel;
	}

	/**
	 * Liefert die Datenbank-ID der Kopplung.
	 * 
	 * @return Die Datenbank-ID der Kopplung.
	 */
	public getID() : number {
		return this._id;
	}

	/**
	 * Setzt das Kürzel der Kopplung.
	 * 
	 * @param pKuerzel  Das neue Kürzel der Kopplung.
	 */
	public setKuerzel(pKuerzel : String) : void {
		this._kuerzel = pKuerzel;
	}

	isTranspiledInstanceOf(name : string): boolean {
		return ['de.nrw.schule.svws.core.utils.stundenplanblockung.StundenplanblockungManagerKopplung'].includes(name);
	}

}

export function cast_de_nrw_schule_svws_core_utils_stundenplanblockung_StundenplanblockungManagerKopplung(obj : unknown) : StundenplanblockungManagerKopplung {
	return obj as StundenplanblockungManagerKopplung;
}
